plugins/auth-backend-module-pinniped-provider/src/authenticator.test.ts

```diff
@@ -494,10 +790,132 @@ describe('pinnipedAuthenticator', () => {
     it('gets id token', async () => {
       const refreshResponse = await pinnipedAuthenticator.refresh(
         refreshRequest,
-        implementation,
+        authCtx,
       );
 
       expect(refreshResponse.session.idToken).toBe(idToken);
     });
+
+    it('refreshes oidc metadata after a failed fetch', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, ctx) =>
+            res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/json'),
+              ctx.json(issuerMetadata),
+            ),
+        ),
+      );
+
+      const response = await pinnipedAuthenticator.refresh(
+        refreshRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
+      );
+      expect(response.session.accessToken).toEqual('accessToken');
+    });
+
+    it('caches oidc metadata after a success', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
+      let supervisorCalls: number = 0;
+
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, ctx) => {
+            supervisorCalls += 1;
+            return res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/json'),
+              ctx.json(issuerMetadata),
+            );
+          },
+        ),
+      );
+
+      await pinnipedAuthenticator.refresh(
+        refreshRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
+      );
+      await pinnipedAuthenticator.refresh(
+        refreshRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
+      );
+
+      expect(supervisorCalls).toEqual(1);
```

Comment:
same critiques as above for this spec -- don't need to start with a failure case

Reply:
Makes sense. Done.